### PR TITLE
Populate intersection type's attached functions in common path

### LINKED ISSUE #31974: The resolved ReturnTypeDescriptor of langlib method is incorrect
**Description:**
Consider the following source code where try to implement the `toString()`  method declaration of runtime:StackFrame object type symbol.  https://github.com/ballerina-platform/ballerina-lang/blob/d4eb62a39664bbee06b2a624751bd453d3a5c9c2/langlib/lang.runtime/src/main/ballerina/runtime.bal#L67-L76

```
import ballerina/lang.runtime;

function test() {
    var myObj = object runtime:StackFrame {
        
    };
}

```
When I access the particular MethodSymbol though methods() API in ObjectTypeSymbol the resolved symbol does not have a return type descriptor as depicted below. 

**Steps to reproduce:**
Follow above steps

**Affected Versions:**
slbeta3-snapshot

### PR DESCRIPTION
**Pull request by pcnfernando**
Fixes #31974

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/object/object_langlib_function_invocation_negative.bal [hunk 1]
```diff
@@ -58,3 +58,9 @@ public class DynamicListenerImpl {
 public readonly class StackFrameImpl {
     *runtime:StackFrame;
 }
+
+function testAnonObjectImpl() {
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/object/object_langlib_function_invocation_negative.bal:62]: Is this correct file for this test? There's no langlib function call, right?

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java [hunk 2]
```diff
@@ -664,7 +664,14 @@ public void visit(BLangFunction astFunc) {
         birFunc.argsCount = astFunc.requiredParams.size()
                 + (astFunc.restParam != null ? 1 : 0) + astFunc.paramClosureMap.size();
         if (astFunc.flagSet.contains(Flag.ATTACHED) && typeDefs.containsKey(astFunc.receiver.getBType().tsymbol)) {
-            typeDefs.get(astFunc.receiver.getBType().tsymbol).attachedFuncs.add(birFunc);
+            BType receiverType = astFunc.receiver.getBType();
+            if (receiverType.tag == TypeTags.OBJECT && ((BObjectType) receiverType).immutableType != null) {
+                BTypeSymbol typeSymbol = ((BObjectType) receiverType).immutableType.effectiveType.tsymbol;
+                if (typeDefs.containsKey(typeSymbol)) {
+                    typeDefs.get(typeSymbol).attachedFuncs.add(birFunc);
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java:671]: I'm not sure if this is the correct fix. Shouldn't the fix be to add the functions for the intersection type to org.wso2.ballerinalang.compiler.tree.BLangPackage#functions?

Comment (pcnfernando) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java:671]: The function is anyway available in the BLangPackage#functions. The issue is that the function does not get populated for the BIRTypeDefinition instance for the Intersection type

Comment (pcnfernando) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java:671]: Had an offline chat, seems we have to duplicate the function for the Intersection type. Will do the necessary changes

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java [hunk 3]
```diff
@@ -1566,32 +1561,6 @@ private void readAndSetEnumSymbol(BUnionType unionType, long flags) throws IOExc
             unionType.tsymbol = new BEnumSymbol(members, flags, names.fromString(enumName), pkgSymbol.pkgID, unionType,
                                                 pkgSymbol, symTable.builtinPos, COMPILED_SOURCE);
         }
-
-        private void populateIntersectionTypeReferencedFunctions(DataInputStream inputStream,
```
[the remark this answers is not in the document]
Comment (pcnfernando) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java]: No, reverted the update

File: tests/jballerina-unit-test/src/test/resources/test-src/object/object_langlib_function_invocation_negative.bal (hunk 1 above)
Comment (pcnfernando) [reply tests/jballerina-unit-test/src/test/resources/test-src/object/object_langlib_function_invocation_negative.bal]: Removed the test as the test scenario is already covered.

**Review by MaryamZi — APPROVED** [review]

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**